src/WarcraftLegacies.Source/Quests/Zandalar/QuestZandalarOutpost.cs

```diff
@@ -0,0 +1,54 @@
+﻿using System.Collections.Generic;
+using MacroTools.Extensions;
+using MacroTools.FactionSystem;
+using MacroTools.ObjectiveSystem.Objectives.FactionBased;
+using MacroTools.ObjectiveSystem.Objectives.UnitBased;
+using MacroTools.QuestSystem;
+using static War3Api.Common;
+
+namespace WarcraftLegacies.Source.Quests.Zandalar
+{
+  /// <summary>
+  /// The Trolls can establish an outpost in the Echo Isles.
+  /// </summary>
+  public sealed class QuestZandalarOutpost : QuestData
+  {
+    private readonly List<unit> _rescueUnits;
+    private readonly ObjectiveAnyUnitInRect _enterZandalarRegion;
+
+    /// <summary>
+    /// Initializes a new instance of the <see cref="QuestZandalar"/>.
+    /// </summary>
+    public QuestZandalarOutpost() : base("Zandalar Outpost",
+      "The trolls of the echo islands have said we can set up an outpost on one of the islands.",
+      @"ReplaceableTextures\CommandButtons\BTNtrollmedium.blp")
+    {
+      AddObjective(_enterZandalarRegion = new ObjectiveAnyUnitInRect(Regions.Zandalari_Echo_Unlock, "Zandalar Outpost", true));
+      AddObjective(new ObjectiveSelfExists());
+      Required = true;
+      ResearchId = Constants.UPGRADE_VQ01_QUEST_COMPLETED_RATCHET_PORT;
+      _rescueUnits = Regions.Zandalari_Echo_Unlock.PrepareUnitsForRescue(RescuePreparationMode.HideNonStructures,
+        filterUnit => filterUnit.GetTypeId() != FourCC("ngme"));
```

Comment:
Do you actually need this filter? Is there a Goblin Merchant around?

Reply:
Missed when copying from Ratchet quest as template. Didn't realise that's what this line was doing. Removed.